python/lsst/jointcal/gtransfo.cc

```diff
@@ -49,6 +49,11 @@ void declareGtransfo(py::module &mod) {
             "inputframe"_a, "inscribed"_a);
     cls.def("getNpar", &Gtransfo::getNpar);
     cls.def("offsetParams", &Gtransfo::offsetParams);
+    cls.def("computeDerivative", [](Gtransfo const &self, Point const &where, const double step) {
+        GtransfoLin derivative;
+        self.computeDerivative(where, derivative, step);
+        return derivative;
```

Comment:
I think this pattern (create a `GtransfoLin`, pass it in and then return it) indicates that the `computeDerivative` method should have an API that returns a new derivative object.

Reply:
The existing C++ uses of `computeDerivative` almost all reuse an existing array for the output. We could look at those and see if they are worth turning into what you suggest, but that's not for this ticket.